### PR TITLE
Use API to select Kubernetes endpoint by name

### LINKED ISSUE #15: Extension: API's to determine if k8s endpoint is setup

### PR DESCRIPTION
**Pull request by wsf11**
Selects Kubernetes endpoint by name, and saves it by ID. This will allow us to use the ID later in the build/release files. Close #15

### CODE REVIEW

File: generators/app/prompt.js [hunk 1]
```diff
@@ -247,6 +247,26 @@ function azureSubList(obj) {
...
+function kubeEndpointList(obj) {
+   return {
+      name: `kubeEndpoint`,
+      type: `list`,
+      store: true,
+      message: `Which Kubernetes Endpoint would you like to use?`,
+      choices: util.getKubeEndpoint,
+      validate: util.validateKubeEndpoint,
+      when: answers => {
+         var result = obj.options.kubeEndpoint === undefined;
```
Comment (wesyao) [inline generators/app/prompt.js]: -- [nit]: let

File: generators/app/utility.js [hunk 2]
```diff
@@ -822,6 +826,47 @@ function getAzureSubs(answers) {
    });
 }
 
+function getKubeEndpoint(answers) {
+   "use strict";
+
+   var token = encodePat(answers.pat);
```
Comment (wesyao) [inline generators/app/utility.js]: [nit]: let

File: generators/app/utility.js [hunk 3]
```diff
@@ -305,6 +305,10 @@ function validateConfigUpdate(input){
    return validateRequired(valid, 'You must configure your Kubernetes endpoint before you answer "Yes". Check docs for more info ');
 }
 
+function validateKubeEndpoint(input) {
+   return validateRequired(input, `You must provide a Kubernetes Service Endpoint!`);
+}
+
 function tokenize(input, nvp) {
    for (var key in nvp) {
```
Comment (wesyao) [inline generators/app/utility.js:313]: -- [nit]: let

File: generators/app/utility.js [hunk 4]
```diff
@@ -822,6 +826,47 @@ function getAzureSubs(answers) {
...
 }
 
+function getKubeEndpoint(answers) {
+   "use strict";
+
+   var token = encodePat(answers.pat);
+   let account_name = answers.tfs;
+   let project = answers.applicationName;
+
+   var options = {
```
Comment (wesyao) [inline generators/app/utility.js]: -- [nit]: let

File: generators/app/utility.js [hunk 5]
```diff
@@ -822,6 +826,47 @@ function getAzureSubs(answers) {
...
+   };
+
+   return new Promise(function (resolve, reject) {
+      request(options, function (e, response, body) {
+         if (e) {
+            reject(e);
+            return;
+         }
+
+         var obj = JSON.parse(body);
```
Comment (wesyao) [inline generators/app/utility.js]: -- [nit]: let

File: generators/app/utility.js [hunk 6]
```diff
@@ -822,6 +826,47 @@ function getAzureSubs(answers) {
...
+
+         var result = [];
```
Comment (wesyao) [inline generators/app/utility.js]: -- [nit]: let

File: generators/app/utility.js [hunk 7]
```diff
@@ -822,6 +826,47 @@ function getAzureSubs(answers) {
    });
 }
 
+function getKubeEndpoint(answers) {
+   "use strict";
+
+   var token = encodePat(answers.pat);
+   let account_name = answers.tfs;
```
Comment (wesyao) [inline generators/app/utility.js]: accountName, i think the underscore format is a python thing.

File: generators/app/utility.js [hunk 8]
```diff
@@ -822,6 +826,47 @@ function getAzureSubs(answers) {
...
+   let project = answers.applicationName;
```
Comment (wesyao) [inline generators/app/utility.js]: projectName?

**Review by wesyao — CHANGES_REQUESTED** [review]
looks good, just some minor fixes

File: generators/app/utility.js (hunk 7 above)
Comment (wsf11) [reply generators/app/utility.js]: You right

File: generators/app/utility.js (hunk 3 above)
Comment (wsf11) [reply generators/app/utility.js:313]: What were you talking about here? key?

**Review by wesyao — APPROVED** [review]

**Pull request merged**